New API for Aruco in OpenCV

Use the new API present in OpenCV 4.x for accessing aruco::DetectorParameters.

Upstream-Status: Pending

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

diff --git a/src/track_marker.cpp b/src/track_marker.cpp
index c0586e47..b396a3ee 100644
--- a/src/track_marker.cpp
+++ b/src/track_marker.cpp
@@ -64,8 +64,17 @@ TrackMarkerNode::TrackMarkerNode(const rclcpp::NodeOptions & options)
   // Default dictionary is cv::aruco::DICT_6X6_250
   int dict_id = this->declare_parameter("dictionary", 10);
 
+  #if CV_VERSION_MAJOR > 4 || CV_VERSION_MAJOR == 4 && CV_VERSION_MINOR >= 7
+  detector_params_ = cv::makePtr<cv::aruco::DetectorParameters>();
+  #else
   detector_params_ = cv::aruco::DetectorParameters::create();
+  #endif
+
+  #if CV_VERSION_MAJOR > 4 || CV_VERSION_MAJOR == 4 && CV_VERSION_MINOR >= 7
+  dictionary_ = cv::makePtr<cv::aruco::Dictionary>(cv::aruco::getPredefinedDictionary(dict_id));
+  #else
   dictionary_ = cv::aruco::getPredefinedDictionary(dict_id);
+  #endif
 
   // Setup lazy subscriber using publisher connection callback
   rclcpp::PublisherOptions pub_options;
